lib/crypto/tests.py

```diff
@@ -205,6 +213,44 @@ def test_sign_file_multi_package(self):
             finally:
                 amo.utils.rm_local_tmp_dir(folder)
 
+    def test_call_signing(self):
+        packaged.call_signing(self.file_, 'endpoint_url')
+        assert self.requests_post_calls == [
+            ('endpoint_url',
+             settings.SIGNING_SERVER_TIMEOUT,
+             {'addon_id': 'xxxxx'},
+             {'file': (u'mozilla.sf',
+                       u'Signature-Version: 1.0\nMD5-Digest-Manifest: '
+                       u'//axi91xGZwlaVjKiw9xuw==\nSHA1-Digest-Manifest: '
+                       u'ep9V9fWlCso0PZUcnM60watGvrM=\n\n')})]
+
+    def test_call_signing_too_long_guid_bug_1203365(self):
+        long_guid = 'x' * 65
+        hashed = hashlib.sha256(long_guid).hexdigest()
+        self.addon.update(guid=long_guid)
+        packaged.call_signing(self.file_, 'endpoint_url')
+        assert self.requests_post_calls == [
+            ('endpoint_url',
+             settings.SIGNING_SERVER_TIMEOUT,
+             {'addon_id': hashed},  # Truncated to 64 chars.
+             {'file': (u'mozilla.sf',
+                       u'Signature-Version: 1.0\nMD5-Digest-Manifest: '
+                       u'//axi91xGZwlaVjKiw9xuw==\nSHA1-Digest-Manifest: '
+                       u'ep9V9fWlCso0PZUcnM60watGvrM=\n\n')})]
+
+    def test_get_id_short_guid(self):
+        assert len(self.addon.guid) <= 64
+        assert len(packaged.get_id(self.addon)) <= 64
+        assert packaged.get_id(self.addon) == self.addon.guid
```

Comment:
I'd add a test with an ID that's exactly 64 characters, just in case of fence post errors.